mlrun/model.py

```diff
@@ -737,8 +737,14 @@ def validate_notification_params(self):
             self.kind
         ).get_notification()
 
-        secret_params = self.secret_params
-        params = self.params
+        secret_params = self.secret_params or {}
+        params = self.params or {}
+
+        # skip the secret_params validation if it contains the key "secret" since the parameters are securely handled
+        # and should not be validated in this case
+        params_secret = secret_params.get("secret", "")
+        if params_secret:
+            return
```

Comment:
I think the heuristic can be improved. We should also verify that this is the only param in the dict because if there are others than they will be ignored. I think we should be explicit about it. (That is IIUC the behavior @quaark )

Reply:
Yes I agree that's a good idea. 
The error message should also be indicative and explain this mutual exclusion